### DISCUSSION TITLE
How to run mqt-qecc on google colab

### DISCUSSION THREAD

**Discussion by hardik954** [cda-tum/mqt-qecc#223] [Q&A]
I want to run the "[test_analog_tannergraph_decoding.py] present in the test.python.analog_info in GOOGLE COLAB folder but how do I do it? I have cloned the repo installed all the requirements yet i can't do it. Can someone please help me

**Comment by burgholzer** [comment]
Hey 👋🏼 
thanks for opening this discussion. 

once you cloned the repository and created a virtual environment (which is highly recommended), you simply install the project and call pytest, i.e., 
```bash
pip install -ve ".[test]"
pytest
```
This will install the `mqt.qecc` package (including the test dependencies) and run all the tests. 
If you just want to run the tests from the `test_analog_tannergraph_decoding.py` file, then replace the last line with
```bash
pytest test/python/analog_info/test_analog_tannergraph_decoding.py
```

> [!NOTE]
> Installing the project requires CMake >= 3.19 and a C++17-compatible compiler.

Detailed instructions on how to set up a development environment are available in the project documentation: https://mqt.readthedocs.io/projects/qecc/en/latest/DevelopmentGuide.html

Hope this resolves your issues. Let us know if anything else goes wrong.
